src/proxy/inbound_passthrough.rs

```diff
@@ -108,8 +108,14 @@ impl InboundPassthrough {
             };
             // Spoofing the source IP only works when the destination or the source are on our node.
             // In this case, the source and the destination might both be remote, so we need to disable it.
+            
+            // The destination zTunnel is now the client zTunnel. We can't impersonate the client,
+            // so use the zTunnel's own IP.
             oc.pi.cfg.enable_original_source = Some(false);
-            return oc.proxy_to(inbound, source.ip(), orig, false).await;
+            let Some(proxy_ip) = oc.pi.cfg.local_ip else {
+               return Err(Error::UnknownSource(source.ip())); 
```

Comment:
Probably not the best/most descriptive error to use.

I saw the DNS stuff leveraged "local-ip-address" package. Maybe config should fallback to that if the env isn't supplied?